xsdata/formats/dataclass/serializers/mixins.py

```diff
@@ -816,7 +819,7 @@ def convert_elements(
         Yields:
             An iterator of sax events.
         """
-        if collections.is_array(value):
+        if collections.is_array(value) or isinstance(value, Generator):
```

Comment:
Let's move this check inside the collections.is_array util method please!

Reply:
Done!